.circleci/config.yml

```diff
@@ -24,6 +24,7 @@ jobs:
       - OPENBLAS_NUM_THREADS: 2
       - CONDA_ENV_NAME: testenv
       - LOCK_FILE: build_tools/circle/doc_min_dependencies_linux-64_conda.lock
+      - WARNINGS_AS_ERRORS: 'false'
```

Comment:
maybe worth following other environment variable with the `SKLEARN_` prefix?

So something like this `SKLEARN_DOC_BUILD_WARNINGS_AS_ERRORS` to be explicit?

Maybe worth adding to the documented variables here in a separate PR:
https://scikit-learn.org/dev/computing/parallelism.html#environment-variables

In particular this may be worth mentioning that this is different from `sphinx-build -W` that also turns some warnings into errors.